notion-server/src/ethereum/mod.rs

```diff
@@ -466,12 +542,102 @@ impl TryFrom<&args::ShardTransaction> for ShardTransaction {
     }
 }
 
+#[derive(Default, Debug)]
+struct BeaconBlock {
+    cross_links: HashMap<ShardChainIndex, CrossLink>,
+}
+impl From<args::BeaconBlock> for BeaconBlock {
+    fn from(beacon_block_args: args::BeaconBlock) -> Self {
+        let mut cross_links: HashMap<ShardChainIndex, CrossLink> = HashMap::new();
+        for (k, v) in beacon_block_args.cross_links {
+            let k = ShardChainIndex(k);
+            let v = CrossLink::from(v);
+            cross_links.insert(k, v);
+        }
+        Self { cross_links }
+    }
+}
+
+#[derive(Default, Debug, Clone)]
+struct CrossLink {
+    execution_environment_states: HashMap<EeIndex, Vec<u8>>,
+}
+impl From<args::CrossLink> for CrossLink {
+    fn from(cross_link_args: args::CrossLink) -> Self {
+        let mut execution_environment_states: HashMap<EeIndex, Vec<u8>> = HashMap::new();
+        for (k, v) in cross_link_args.execution_environment_states {
+            let k = EeIndex(k);
+            execution_environment_states.insert(k, v);
+        }
+        Self {
+            execution_environment_states,
+        }
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use super::*;
     use hex::FromHex;
     use std::path::Path;
 
+    #[test]
+    fn can_create_and_get_beacon_blocks() {
```

Comment:
I recommend breaking this up into smaller tests. Perhaps one for creating and a separate one for getting?

Since you're in the same module, you can directly inspect the struct's internal state.

Reply:
IMO it's easier/better as 1 test, since to test retrieval you need to add it first, so combining demonstrates that get/retrieve both work via the main interface.  Are you suggesting breaking it into two and creating each by modifying internal state vs using the external interface?

Reply:
>Are you suggesting breaking it into two and creating each by modifying internal state vs using the external interface?

Yep! That's exactly what I'm suggesting. If one half (add or get) is broken, splitting into smaller tests makes it easier to identify.

If generating the correct internal state is particularly painful, I don't mind leaving it as is.

Reply:
Fair enough. I think this test will be fairly easy to read and track down issues in the event that it fails.  I like using the external interface for this because it better demonstrates that the functions work in a real world sense, and it simplifies the writing of the test.

For now I think I'll leave as is, but I appreciate the thoughts on this, think your points are valid.